crates/chia-consensus/src/gen/condition_tools.rs

```diff
@@ -0,0 +1,194 @@
+            n if n >= 0x8000_0000_0000_u64 => 1,
+            n if n >= 0x0080_0000_0000_u64 => 2,
+            n if n >= 0x8000_0000_u64 => 3,
+            n if n >= 0x0080_0000_u64 => 4,
+            n if n >= 0x8000_u64 => 5,
+            n if n >= 0x80_u64 => 6,
+            n if n > 0 => 7,
+            _ => 8,
+        };
+        Bytes::new(amount_bytes[start..].to_vec())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::allocator::make_allocator;
+    use crate::consensus_constants::TEST_CONSTANTS;
+    use clvmr::chia_dialect::LIMIT_HEAP;
+    use clvmr::Allocator;
+    use hex_literal::hex;
+    use rstest::rstest;
+
+    #[test]
+    fn test_validate_u64() {
+        let mut a: Allocator = make_allocator(LIMIT_HEAP);
+        for v in 0..10000 {
+            let ptr = a.new_small_number(v).expect("valid u64");
+            assert_eq!(a.atom(ptr).as_ref(), u64_to_bytes(v as u64).as_slice());
+        }
+        for v in 18_446_744_073_709_551_615_u64 - 1000..18_446_744_073_709_551_615 {
+            let ptr = a.new_number(v.into()).expect("valid u64");
+            assert_eq!(a.atom(ptr).as_ref(), u64_to_bytes(v).as_slice());
+        }
+    }
+
+    #[rstest]
+    #[case(AGG_SIG_PARENT, b"parent_message", hex!("4444444444444444444444444444444444444444444444444444444444444444").into(), hex!("3333333333333333333333333333333333333333333333333333333333333333").into(), 10000)]
+    #[case(AGG_SIG_PUZZLE, b"puzzle_message", hex!("4444444444444444444444444444444444444444444444444444444444444444").into(), hex!("3333333333333333333333333333333333333333333333333333333333333333").into(), 261)]
+    #[case(AGG_SIG_AMOUNT, b"amount_message", hex!("4444444444444444444444444444444444444444444444444444444444444444").into(), hex!("3333333333333333333333333333333333333333333333333333333333333333").into(), 100_000_000_005)]
+    #[case(AGG_SIG_PUZZLE_AMOUNT, b"puzzle_amount_message", hex!("4444444444444444444444444444444444444444444444444444444444444444").into(), hex!("3333333333333333333333333333333333333333333333333333333333333333").into(), 410)]
+    #[case(AGG_SIG_PARENT_AMOUNT, b"parent_amount_message", hex!("4444444444444444444444444444444444444444444444444444444444444444").into(), hex!("3333333333333333333333333333333333333333333333333333333333333333").into(), 909)]
+    #[case(AGG_SIG_PARENT_PUZZLE, b"parent_puzzle_message", hex!("4444444444444444444444444444444444444444444444444444444444444444").into(), hex!("3333333333333333333333333333333333333333333333333333333333333333").into(), 10_061_997)]
+    #[case(AGG_SIG_ME, b"me_message", hex!("4444444444444444444444444444444444444444444444444444444444444444").into(), hex!("3333333333333333333333333333333333333333333333333333333333333333").into(), 1303)]
+    fn test_make_aggsig_final_message(
+        #[case] opcode: ConditionOpcode,
+        #[case] msg: &[u8],
+        #[case] parent_id: Vec<u8>,
+        #[case] puzzle_hash: Vec<u8>,
```

Comment:
every test case uses the same values for `parent_id` and `puzzle_hash`. I don't think you need to make these parameters to the test.

I think you could make `msg` a constant in the tests as well, since you don't need to test different values of it (I think).

Reply:
Done